lib/remote/configstageshandler.cpp

```diff
@@ -97,6 +100,13 @@ void ConfigStagesHandler::HandlePost(
 {
 	namespace http = boost::beast::http;
 
+	if (m_RunningPackageUpdates.exchange(true)) {
+		return HttpUtility::SendJsonError(response, params, 423,
+			"Conflicting request, there is already an ongoing package update in progress. Please try it again later.");
+	}
+
+	auto resetPackageUpdates (Shared<Defer>::Make([]() { ConfigStagesHandler::m_RunningPackageUpdates.store(false); }));
```

Comment:
This should be done as late as possible, in particular after all the request validation that follows. These checks can be done safely before that lock is obtained.